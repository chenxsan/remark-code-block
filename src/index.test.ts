import { unified } from 'unified'
import type { Preset } from 'unified'
import remarkParse from 'remark-parse'
import remarkRehype from 'remark-rehype'
import rehypeStringify from 'rehype-stringify'
import { remarkCodeBlock } from '.'
import { describe, it, expect } from 'vitest'

import js from 'refractor/lang/javascript'
import html from 'refractor/lang/markup'

describe('remarkCodeBlock', () => {
  const processor = unified()
    .use(remarkParse)
    .use(remarkCodeBlock, {
      enableLineNumbers: true,
    })
    .use(remarkRehype as Preset)
    .use(rehypeStringify)

  remarkCodeBlock.register(js)
  remarkCodeBlock.register(html)

  it('should highlight code block', () => {
    processor.process(
      `
\`\`\`javascript
console.log('remark code block');


console.log('remark code block');
\`\`\`
`,
      function (err, file) {
        expect(err).toBeNull()
        if (!err) {
          expect(file?.value).toContain(`token punctuation`)

          expect(file?.value).toMatchSnapshot()
        }
      }
    )
  })
  it('should add data-filename to code element', () => {
    processor.process(
      `
\`\`\`html filename=src/pages/hello.html
<p>hello world</p>
\`\`\`
`,
      (err, file) => {
        if (!err) {
          expect(file?.value).toContain('data-filename="src/pages/hello.html"')
          expect(file?.value).toContain('language-html')
          expect(file?.value).toContain('data-lang="html"')

          expect(file?.value).toMatchSnapshot()
        }
      }
    )
  })
  it('should add data-line-number', () => {
    processor.process(
      `
\`\`\`html filename=src/pages/hello.html
<p>hello world</p>

<div>html 5</div>
\`\`\`
`,
      (err, file) => {
        if (!err) {
          expect(file?.value).toContain('data-line-number="1"')
        }
      }
    )
  })
  it('should highlight lines', () => {
    processor.process(
      `
\`\`\`html lines=[1,3]
<p>hello world</p>

<div>html 5</div>
\`\`\`
`,
      (err, file) => {
        if (!err) {
          expect(file?.value).toContain('data-line-highlight')
        }
      }
    )
  })
})
