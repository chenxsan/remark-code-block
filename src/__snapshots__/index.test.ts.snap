// Vitest Snapshot v1

exports[`remarkCodeBlock > should add data-filename to code element 1`] = `"<pre><code class=\\"language-html\\" data-lang=\\"html\\" data-filename=\\"src/pages/hello.html\\"><span data-line-number=\\"1\\"><span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&#x3C;</span>p</span><span class=\\"token punctuation\\">></span></span>hello world<span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&#x3C;/</span>p</span><span class=\\"token punctuation\\">></span></span></span></code></pre>"`;

exports[`remarkCodeBlock > should highlight code block 1`] = `
"<pre><code class=\\"language-javascript\\" data-lang=\\"javascript\\"><span data-line-number=\\"1\\">console<span class=\\"token punctuation\\">.</span><span class=\\"token function\\">log</span><span class=\\"token punctuation\\">(</span><span class=\\"token string\\">'remark code block'</span><span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">;</span>
</span><span data-line-number=\\"2\\">
</span><span data-line-number=\\"3\\">
</span><span data-line-number=\\"4\\">console<span class=\\"token punctuation\\">.</span><span class=\\"token function\\">log</span><span class=\\"token punctuation\\">(</span><span class=\\"token string\\">'remark code block'</span><span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">;</span></span></code></pre>"
`;
